app/src/main/java/org/kiwix/kiwixmobile/intro/IntroActivity.kt

```diff
@@ -0,0 +1,117 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.intro
+
+import android.os.Bundle
+import android.os.Handler
+import android.view.View
+import androidx.core.view.isVisible
+import androidx.viewpager.widget.ViewPager
+import kotlinx.android.synthetic.main.activity_intro.*
+import kotlinx.android.synthetic.main.item_intro_2.airplane
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.Intents.internal
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import org.kiwix.kiwixmobile.zim_manager.SimplePageChangeListener
+import java.util.Timer
+import java.util.TimerTask
+import javax.inject.Inject
+
+class IntroActivity : BaseActivity(), IntroContract.View {
+
+  companion object {
+    private const val timerDelay: Long = 0
+    private const val timerPeriod: Long = 2000
+    private const val animationDuration: Long = 800
+  }
+
+  private val handler = Handler()
+  private val timer = Timer()
+
+  @Inject
+  internal lateinit var presenter: IntroContract.Presenter
+  private var currentPage = 0
+  private lateinit var views: Array<View>
+
+  override fun injection(coreComponent: CoreComponent) {
+    this.kiwixActivityComponent.inject(this)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_intro)
+    get_started.setOnClickListener { startMainActivity() }
+    views = arrayOf(
+      layoutInflater.inflate(R.layout.item_intro_1, view_pager, false),
+      layoutInflater.inflate(R.layout.item_intro_2, view_pager, false)
+    )
+    view_pager.run {
+      adapter = IntroPagerAdapter(views)
+      addOnPageChangeListener(SimplePageChangeListener(::updateView, ::handleDraggingState))
+    }
+    tab_indicator.setViewPager(view_pager)
+    timer.schedule(object : TimerTask() {
+      override fun run() {
+        handler.post {
+          if (currentPage == views.size) currentPage = 0
+          view_pager.setCurrentItem(currentPage++, true)
+        }
+      }
+    }, timerDelay, timerPeriod)
+    views.forEach {
+      it.setOnClickListener { dismissAutoRotate() }
+    }
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    handler.removeCallbacksAndMessages(null)
+    timer.cancel()
+    views.forEach {
+      it.setOnClickListener(null)
+    }
+  }
+
+  private fun startMainActivity() {
+    dismissAutoRotate()
+    startActivity(internal(CoreMainActivity::class.java))
+    presenter.setIntroShown()
+    finish()
+  }
+
+  private fun updateView(position: Int) {
+    airplane.isVisible = position == 1
+    if (position == 1)
```

Comment:
this should have braces, we should only omit braces when `if` is used as an expression eg

```
val myInt = if(condition) 1 else 2

fun example(aBool:Boolean){
return if (aBool) 1 else 2
}
```